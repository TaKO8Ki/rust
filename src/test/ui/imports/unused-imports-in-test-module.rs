#![deny(unused_imports)]

use std::io::BufRead; //~ ERROR unused import: `std::io::BufRead`

fn a() {}
fn b() {}

mod test {
    use super::a;  //~ ERROR unused import: `super::a`

    fn foo() {
        use crate::b;  //~ ERROR unused import: `crate::b`
    }
}

mod test_a {
    use super::a;  //~ ERROR unused import: `super::a`

    fn foo() {
        use crate::b;  //~ ERROR unused import: `crate::b`
    }
}

fn main() {}
