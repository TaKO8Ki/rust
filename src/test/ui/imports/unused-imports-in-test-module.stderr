error: unused import: `std::io::BufRead`
  --> $DIR/unused-imports-in-test-module.rs:3:5
   |
LL | use std::io::BufRead;
   |     ^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/unused-imports-in-test-module.rs:1:9
   |
LL | #![deny(unused_imports)]
   |         ^^^^^^^^^^^^^^

error: unused import: `super::a`
  --> $DIR/unused-imports-in-test-module.rs:9:9
   |
LL |     use super::a;
   |         ^^^^^^^^
   |
help: consider adding a `#[cfg(test)]` to the module named `test` in which this import is
  --> $DIR/unused-imports-in-test-module.rs:8:1
   |
LL | mod test {
   | ^^^^^^^^

error: unused import: `crate::b`
  --> $DIR/unused-imports-in-test-module.rs:12:13
   |
LL |         use crate::b;
   |             ^^^^^^^^
   |
help: consider adding a `#[cfg(test)]` to the module named `test` in which this import is
  --> $DIR/unused-imports-in-test-module.rs:8:1
   |
LL | mod test {
   | ^^^^^^^^

error: aborting due to 3 previous errors

